internal/workflow.go

```diff
@@ -1530,10 +1530,26 @@ func WithLocalActivityOptions(ctx Context, options LocalActivityOptions) Context
 
 	opts.ScheduleToCloseTimeout = options.ScheduleToCloseTimeout
 	opts.StartToCloseTimeout = options.StartToCloseTimeout
-	opts.RetryPolicy = options.RetryPolicy
+	opts.RetryPolicy = applyRetryPolicyDefaultsForLocalActivity(options.RetryPolicy)
 	return ctx1
 }
 
+func applyRetryPolicyDefaultsForLocalActivity(policy *RetryPolicy) *RetryPolicy {
```

Comment:
where do we enforce that it only retry up to ScheduleToCloseTimeout? Is the comment on LocalActivityOption accurate?

Reply:
It's at the LA [task level](https://github.com/temporalio/sdk-go/blob/0550203f48dd2e84f0db6d0374b1c58dcc9b33a5/internal/internal_event_handlers.go#L530), we calculate it right away and set it as a context deadline.